payments/repo.go

```diff
@@ -20,11 +20,25 @@ func (p *PaymentRepo) Get(id uint) (*database.Payment, error) {
 	return &payment, err
 }
 
-func (p *PaymentRepo) GetAllByBudgetCategoryID(id uint) ([]*database.Payment, error) {
-	var payments []*database.Payment
+func (p *PaymentRepo) GetAllByBudgetID(id uint) (*[]database.Payment, error) {
```

Comment:
Bad developer, bad!

```
[]Thing
```

Is the only one we use, as it has known characteristics. Arrays/slices are cheap, and (linearly) stack allocated, no need for pointer indirection.